File: zad2.py
class Car:
    def __init__(self, marka, model, typ, cena_za_dzien):
        self.marka = marka
        self.model = model
        self.typ = typ
        self.cena_za_dzien = cena_za_dzien

    def __str__(self):
        return f"{self.marka} {self.model} ({self.typ})"

class Customer:
    def __init__(self, imie_i_nazwisko, numer_prawa_jazdy):
        self.imie_i_nazwisko = imie_i_nazwisko
        self.numer_prawa_jazdy = numer_prawa_jazdy

class Rental:
    def __init__(self, samochod, klient, liczba_dni):
        self.samochod = samochod
        self.klient = klient
        self.liczba_dni = liczba_dni
        self.koszt_calkowity = samochod.cena_za_dzien * liczba_dni

    def __str__(self):
        return f"{self.klient.imie_i_nazwisko} wypożyczył(a) {self.samochod.marka} {self.samochod.model} na {self.liczba_dni} dni. Koszt: {self.koszt_calkowity:.2f} zł"

samochod1 = Car("Toyota", "Corolla", "Sedan", 120)
klient1 = Customer("Anna Kowalska", "XYZ123456")
wypozyczenie1 = Rental(samochod1, klient1, 3)

print(samochod1)
print(wypozyczenie1)

File: zad2.cs
using System;

class Car
{
    public string Marka { get; }
    public string Model { get; }
    public string Typ { get; }
    public decimal CenaZaDzien { get; }

    public Car(string marka, string model, string typ, decimal cenaZaDzien)
    {
        Marka = marka;
        Model = model;
        Typ = typ;
        CenaZaDzien = cenaZaDzien;
    }

    public override string ToString()
    {
        return $"{Marka} {Model} ({Typ})";
    }
}

class Customer
{
    public string ImieINazwisko { get; }
    public string NumerPrawaJazdy { get; }

    public Customer(string imieINazwisko, string numerPrawaJazdy)
    {
        ImieINazwisko = imieINazwisko;
        NumerPrawaJazdy = numerPrawaJazdy;
    }
}

class Rental
{
    public Car Samochod { get; }
    public Customer Klient { get; }
    public int LiczbaDni { get; }
    public decimal KosztCalkowity { get; }

    public Rental(Car samochod, Customer klient, int liczbaDni)
    {
        Samochod = samochod;
        Klient = klient;
        LiczbaDni = liczbaDni;
        KosztCalkowity = samochod.CenaZaDzien * liczbaDni;
    }

    public override string ToString()
    {
        return $"{Klient.ImieINazwisko} wypożyczył(a) {Samochod.Marka} {Samochod.Model} na {LiczbaDni} dni. Koszt: {KosztCalkowity:F2} zł";
    }
}

class Program
{
    static void Main()
    {
        var samochod1 = new Car("Toyota", "Corolla", "Sedan", 120);
        var klient1 = new Customer("Anna Kowalska", "XYZ123456");
        var wypozyczenie1 = new Rental(samochod1, klient1, 3);

        Console.WriteLine(samochod1);
        Console.WriteLine(wypozyczenie1);
    }
}
